### ISSUE TITLE
Feature Request: Add event callback when click line

### ISSUE THREAD

**Issue by Zmnortal** [RexSkz/json-diff-kit#44]
Hello, this library is really convenient, and I am very grateful to the author! 

I am currently using this library at work for some batch update checks. In this process, I found that if I could directly click on the code diff to remove unnecessary diffs, it would be very convenient (I also use the just-diff package to patch the diffs). Currently, there is no way to achieve this requirement, so it's a pity. If possible, I hope to add event callbacks, which would really be helpful! If possible, I am willing  to contribute code. 

Once again, thank you very much to the author.

**Comment by RexSkz** [comment]
PR is always welcome!

Just be curious: I don't know how you map the diff results to the original key or value; if I implement this feature, I can only provide the `DiffResult` object as the callback parameter, which seems not easy to achieve your original goal.

**Comment by Zmnortal** [comment]
Yes, this is indeed a problem that I need to find a way to solve.
My intuitive thought is that the callback will carry information such as the content of the current line and the line number, and then compare it with the original information to determine the specific part of the diff. Since my current requirement is to replace keywords, the diff will only occur within a single line and not span multiple lines. Moreover, the possibility of the diff content being repetitive is quite low, and I can specifically handle this part of the bad case. The line numbers might not align due to different formatters, but in most scenarios, it should be feasible through content. I want to give it a try.

**Comment by RexSkz** [comment]
I see, you just want a feature like this:

```ts
interface RowClickInfo {
  // must have
  diff: [DiffResult, DiffResult];

  // nice to have
  side: 'left' | 'right';
  isLineNumberColumn: boolean;
}

const viewerProps: ViewerProps = {
  onRowClick: (info: RowClickInfo) => {
    const [left, right] = info.diff;
    console.log(left.lineNumber, left.text, left.comma);
  },
};
```

Feel free to submit a PR :)

**Comment by Zmnortal** [comment]
Yes, this is good enough. I will submit this PR I think.
